mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -1936,55 +2064,112 @@ void StartProcess(const std::wstring &command)
     }
 }
 
-#pragma endregion // functions
-
 // =====================================================================
 
-// main body of the mod called every time a taskbar is clicked
-bool OnMouseClick(HWND hWnd, WPARAM wParam, LPARAM lParam, TaskBarAction taskbarAction)
+bool IsTaskbarWindow(HWND hWnd)
 {
     LOG_TRACE();
 
-    if ((GetCapture() != NULL) || (taskbarAction == ACTION_NOTHING))
+    WCHAR szClassName[32];
+    if (GetClassName(hWnd, szClassName, ARRAYSIZE(szClassName)))
+    {
+        return _wcsicmp(szClassName, L"Shell_TrayWnd") == 0 || _wcsicmp(szClassName, L"Shell_SecondaryTrayWnd") == 0;
+    }
+    else
     {
+        LOG_ERROR(L"Failed to get window class name");
         return false;
     }
+}
 
-    // old Windows mouse handling of WM_MBUTTONDOWN message
-    POINT pointerLocation{};
-    if (!GetMouseClickPosition(lParam, pointerLocation))
+bool IsDoubleClick()
+{
+    LOG_TRACE();
+
+    const MouseClick &previousClick = g_mouseClickQueue[-2];
+    const MouseClick &currentClick = g_mouseClickQueue[-1];
+
+    if (previousClick.type != MouseClick::Type::MOUSE || currentClick.type != MouseClick::Type::MOUSE)
     {
         return false;
     }
 
-    // Note: The reason why UIAutomation interface is used is that it reliably returns a className of the element clicked.
-    // If standard Windows API is used, the className returned is always Shell_TrayWnd which is a parrent window wrapping the taskbar.
-    // From that we can't really tell reliably whether user clicked on the taskbar empty space or on some UI element on that taskbar, like
-    // opened window, icon, start menu, etc.
+    // Check if the current event is within the double-click time and distance
+    bool result = abs(previousClick.position.x - currentClick.position.x) <= GetSystemMetrics(SM_CXDOUBLECLK) &&
+                  abs(previousClick.position.y - currentClick.position.y) <= GetSystemMetrics(SM_CYDOUBLECLK) &&
+                  ((currentClick.timestamp - previousClick.timestamp) <= GetDoubleClickTime());
```

Comment:
I assume DPI is relevant here too. Perhaps you can use `GetSystemMetricsForDpi` here.